scene/resources/texture.cpp

```diff
@@ -1087,8 +1087,12 @@ bool AtlasTexture::is_pixel_opaque(int p_x, int p_y) const {
 	if (!atlas.is_valid())
 		return true;
 
-	int x = p_x + region.position.x + margin.position.x;
-	int y = p_y + region.position.y + margin.position.y;
+	int x = p_x + region.position.x - margin.position.x;
+	int y = p_y + region.position.y - margin.position.y;
+
+	// margin edge may outside of atlas
+	if (x < 0 || x >= atlas->get_width()) return false;
```

Comment:
You'll need to amend the commit to add a line break before `return` to match our [style guide](http://docs.godotengine.org/en/latest/community/contributing/code_style_guidelines.html). We don't allow putting condition and statement on the same line for `if`, `for`, etc.

Reply:
Actually nevermind, the style guide is fine with this.